src/battle_bg.c

```diff
@@ -0,0 +1,925 @@
+    {gUnknown_824E858, gUnknown_824EB90, gUnknown_824E410, gUnknown_824E490, gUnknown_824E7DC}
+};
+
+static const struct {
+    u8 unk0;
+    u8 unk1;
+} gUnknown_824EFC4[] = {
+    {1, 11},
+    {2, 14},
+    {3, 13},
+    {4, 15},
+    {5, 16},
+    {6, 17},
+    {7, 18},
+    {8, 10}
+};
+
+UNUSED void sub_800F1E8(void)
+{
+    u8 spriteId;
+
+    ResetSpriteData();
+    spriteId = CreateSprite(&gUnknown_824EFF0, 0, 0, 0);
+    gSprites[spriteId].invisible = TRUE;
+    SetMainCallback2(sub_800F228);
+}
+
+static void sub_800F228(void)
+{
+    AnimateSprites();
+    BuildOamBuffer();
+}
+
+static u8 sub_800F238(u8 a0)
+{
+    int i;
+    for (i = 0; i < NELEMS(gUnknown_824EFC4); i++)
+    {
+        if (a0 == gUnknown_824EFC4[i].unk0)
+            return gUnknown_824EFC4[i].unk1;
+    }
+    return 9;
+}
+
+static void sub_800F260(u16 envId)
+{
+    if (envId >= NELEMS(gUnknown_824EE34))
+        envId = 9;
+    LZDecompressVram(gUnknown_824EE34[envId][0], (void *)BG_CHAR_ADDR(2));
+    LZDecompressVram(gUnknown_824EE34[envId][1], (void *)BG_SCREEN_ADDR(26));
+    LoadCompressedPalette(gUnknown_824EE34[envId][4], 0x20, 0x60);
+}
+
+static void sub_800F2AC(u16 envId)
+{
+    if (envId >= NELEMS(gUnknown_824EE34))
+        envId = 9;
+    LZDecompressVram(gUnknown_824EE34[envId][2], (void *)BG_CHAR_ADDR(1));
+    LZDecompressVram(gUnknown_824EE34[envId][3], (void *)BG_SCREEN_ADDR(28));
+}
+
+UNUSED void sub_800F2EC(u8 envId, const u32 **tilesPtr, const u32 **mapPtr, const u32 **palPtr)
+{
+    if (envId > 9)
+        envId = 9;
+    *tilesPtr = gUnknown_824EE34[envId][0];
+    *mapPtr = gUnknown_824EE34[envId][1];
+    *palPtr = gUnknown_824EE34[envId][4];
+}
+
+void sub_800F324(void)
+{
+    ResetBgsAndClearDma3BusyFlags(FALSE);
+    InitBgsFromTemplates(0, gBattleBgTemplates, NELEMS(gBattleBgTemplates));
+    InitWindows(gUnknown_8248330);
+    DeactivateAllTextPrinters();
+}
+
+void InitBattleBgsVideo(void)
+{
+    EnableInterrupts(INTR_FLAG_VBLANK | INTR_FLAG_VCOUNT | INTR_FLAG_TIMER3 | INTR_FLAG_SERIAL);
+    sub_800F324();
+    SetGpuReg(REG_OFFSET_BLDCNT, 0);
+    SetGpuReg(REG_OFFSET_BLDALPHA, 0);
+    SetGpuReg(REG_OFFSET_BLDY, 0);
+    SetGpuReg(REG_OFFSET_DISPCNT, DISPCNT_MODE_0 | DISPCNT_OBJ_1D_MAP | DISPCNT_OBJ_ON | DISPCNT_WIN0_ON | DISPCNT_OBJWIN_ON);
+}
+
+void LoadBattleMenuWindowGfx(void)
+{
+    TextWindow_SetUserSelectedFrame(2, 0x012, 0x10);
+    TextWindow_SetUserSelectedFrame(2, 0x022, 0x10);
+    gPlttBufferUnfaded[0x5C] = RGB( 9,  9,  9);
+    gPlttBufferUnfaded[0x5D] = RGB( 9,  9,  9);
+    gPlttBufferUnfaded[0x5E] = RGB(31, 31, 31);
+    gPlttBufferUnfaded[0x5F] = RGB( 26,  26,  25);
+    CpuCopy16(&gPlttBufferUnfaded[0x5C], &gPlttBufferFaded[0x5C], 8);
+    if (gBattleTypeFlags & (BATTLE_TYPE_FIRST_BATTLE | BATTLE_TYPE_POKEDUDE))
+    {
+        Menu_LoadStdPalAt(0x70);
+        TextWindow_LoadResourcesStdFrame0(0, 0x030, 0x70);
+        gPlttBufferUnfaded[0x76] = RGB( 0,  0,  0);
```

Comment:
`RGB_BLACK`

Reply:
Keeping it this way because surrounding RGB values are specified with parameters.